### PR TITLE
[FLASK] Bump Snaps packages

### PR DESCRIPTION
**Pull request by Mrtenz**
## Explanation

This bumps the Snaps packages to their latest versions, to add support for the new lifecycle hooks, and fix some audit findings.

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by Mrtenz** [comment]
@metamaskbot update-policies

**Review by hmalik88 — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -30797,14 +30838,14 @@ __metadata:
   languageName: node
   linkType: hard
 
-"semver@npm:^7.0.0, semver@npm:^7.3.2, semver@npm:^7.3.5, semver@npm:^7.3.7, semver@npm:^7.3.8":
-  version: 7.5.0
-  resolution: "semver@npm:7.5.0"
+"semver@npm:^7.0.0, semver@npm:^7.3.2, semver@npm:^7.3.5, semver@npm:^7.3.7, semver@npm:^7.3.8, semver@npm:^7.5.4":
+  version: 7.5.4
```
[the remark this answers is not in the document]
Comment (Mrtenz) [reply yarn.lock:30842]: Sounds good to me. I just ran `yarn dedupe` though 😅

**Comment by bowensanders** [comment]
Manual QA pass ✅

**Comment by Mrtenz** [comment]
@metamaskbot update-policies

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -962,6 +962,8 @@ export default class MetamaskController extends EventEmitter {
         'ExecutionService:unhandledError',
         'ExecutionService:outboundRequest',
         'ExecutionService:outboundResponse',
+        'SnapController:snapInstalled',
+        'SnapController:snapUpdated',
       ],
       allowedActions: [
```
Comment (GuillaumeRx) [inline app/scripts/metamask-controller.js:968]: Although the function that is requiring the `SnapRegistry:update` action is never used at this point, I don't know if it's not a good idea to add `SnapRegistry:update` to the `allowedActions` ?

Comment (Mrtenz) [reply app/scripts/metamask-controller.js:968]: If it's not used, I don't think we should add it?

Comment (Mrtenz) [reply app/scripts/metamask-controller.js:968]: We can always add it in a future PR if it becomes necessary.

**Review by david0xd — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**